feat: number convertor

Daooobg opened the pull request:
This is what i did

Daooobg committed:
feat: number convertor

This is what i did

antoniaat committed:
Merge branch 'main' into 09-numberConvertor

antoniaat commented on 04-dom/exercise/09-numberConvertor.js:
```diff
@@ -0,0 +1,59 @@
+function solve() {
+
+    let optionBinary = document.createElement("OPTION");
```
Why not using `const` here as well?

antoniaat replied:
+When we are creating or taking an element reference I suggest to always put name in the end so to remember that inside we have live DOM element.

antoniaat commented on 04-dom/exercise/09-numberConvertor.js:
```diff
@@ -0,0 +1,59 @@
+    selectMenuTo.appendChild(optionBinary);
+    selectMenuTo.appendChild(optionHexadecimal);
+
+    btn[0].onclick = function () {
```
What is btn[0]? I suggest creating a variable with descriptive name for this.

antoniaat commented on 04-dom/exercise/09-numberConvertor.js:
```diff
@@ -0,0 +1,59 @@
+        return result.join('');
+    }
+
+    function convertToHeXadecimal(num) {
```
Of course this function works fine but if you have the wish you can research for some ways to improve it. I would suggest something like this (not tested, just a suggestion):

```javascript
function convertToHexadecimal(num) {
  // Base case: if num is 0, return '0' (special case)
  if (num === 0) return '0';

  // Define a string containing all hexadecimal characters in order
  const hexChars = '0123456789ABCDEF';

  let result = '';

  // Loop until num becomes 0
  while (num > 0) {
    // Calculate the remainder when dividing by 16 (hexadecimal base)
    const digit = num % 16;

    // Prepend the corresponding hexadecimal character to the result
    result = hexChars[digit] + result;

    // Reduce num by the integer division by 16
    num = Math.floor(num / 16);
  }

  return result;
}
```

antoniaat reviewed (changes requested):
The branch should point to main - please rebase.

Daooobg committed:
Update 09-numberConvertor.js

I made the changes and the function that converts is  changed. I used toString(16) to convert the number and toUpperCase() to make the letters uppercase

Daooobg committed:
Update 09-numberConvertor.js

BREAKING CHANGE

Daooobg committed:
Merge branch 'main' into 09-numberConvertor

Daooobg committed:
Update 09-numberConvertor.js

remove repetitive code

Daooobg committed:
Merge branch '09-numberConvertor' of https://github.com/Daooobg/JS-Advanced-Sep-2023 into 09-numberConvertor

Commit:
Merge branch 'main' into 09-numberConvertor

antoniaat reviewed (approved):
Looks much better! Great job!

Commit:
Merge branch 'main' into 09-numberConvertor

Daooobg committed:
Merge branch 'main' into 09-numberConvertor
